man/man8/zfs.8

```diff
@@ -513,6 +513,11 @@ reducing the total amount of data stored. If a file system has the
 property enabled, duplicate data blocks are removed synchronously. The result
 is that only unique data is stored and common components are shared among files.
 .Pp
+.Bf Em
+WARNING: DO NOT ENABLE DEDUPLICATION UNLESS YOU NEED IT AND KNOW EXACTLY WHAT
+YOU ARE DOING!
```

Comment:
This is pretty scary... how would one learn what they are doing, and if dedup is appropriate or not?  AFAIK, dedup works, despite having rather extreme performance impact in some (perhaps most) cases.  I think we'd be better off being more specific about the downsides rather than including a generic scary message.

Reply:
It is scary, but <a href="https://github.com/zfsonlinux/zfs/blob/682ce104cdd80db4b67eea09eb0a90324c5f98ee/man/man8/zfs.8#L384">I'm just copying it from the old version of the man page</a>. More details about planning and expected issues are in the man page just below this warning.

Reply:
I agree with @ahrens, we can be prescriptive. For now, until special classes are available, I think we can recommend against dedup for HDD pools.

Reply:
@DeHackEd thanks for catching this, but actually I dropped this scary warning intentionally in the mdoc conversion.  The existing text in the "Deduplication" section seemed sufficient.